sdk/identity/Azure.Identity/src/ManagedIdentitySource.cs

```diff
@@ -40,20 +41,33 @@ public virtual async ValueTask<AccessToken> AuthenticateAsync(bool async, TokenR
             return await HandleResponseAsync(async, context, message.Response, cancellationToken).ConfigureAwait(false);
         }
 
-        protected virtual async ValueTask<AccessToken> HandleResponseAsync(bool async, TokenRequestContext context, Response response, CancellationToken cancellationToken)
+        protected virtual async ValueTask<AccessToken> HandleResponseAsync(
+            bool async,
+            TokenRequestContext context,
+            Response response,
+            CancellationToken cancellationToken)
         {
-            using JsonDocument json = async
-                ? await JsonDocument.ParseAsync(response.ContentStream, default, cancellationToken).ConfigureAwait(false)
-                : JsonDocument.Parse(response.ContentStream);
-            if (response.Status == 200)
+            try
             {
-                return GetTokenFromResponse(json.RootElement);
-            }
+                using JsonDocument json = async
+                    ? await JsonDocument.ParseAsync(response.ContentStream, default, cancellationToken).ConfigureAwait(false)
+                    : JsonDocument.Parse(response.ContentStream);
+                if (response.Status == 200)
+                {
+                    return GetTokenFromResponse(json.RootElement);
+                }
 
-            string message = GetMessageFromResponse(json.RootElement);
-            throw async
-                ? await Pipeline.Diagnostics.CreateRequestFailedExceptionAsync(response, message).ConfigureAwait(false)
-                : Pipeline.Diagnostics.CreateRequestFailedException(response, message);
+                var message = GetMessageFromResponse(json.RootElement);
+                throw async
+                    ? await Pipeline.Diagnostics.CreateRequestFailedExceptionAsync(response, message).ConfigureAwait(false)
+                    : Pipeline.Diagnostics.CreateRequestFailedException(response, message);
+            }
+            catch (Exception e)
+            {
```

Comment:
The above throw would also get caught here as well. I don't think that's  what we want.